### PR TITLE
tools: refactor API Bench and support HTTP server

### PR DESCRIPTION
**Pull request by CabinfeverB**
### What problem does this PR solve?

Issue Number: ref #6922

### What is changed and how does it work?

```commit-message
```

### Check List

Tests

no code changed about PD server and PD client
- Manual test (add detailed scripts or steps below)

Code changes

- Has the configuration change
- Has HTTP API interfaces changed (Don't forget to [add the declarative for the new API](https://github.com/tikv/pd/blob/master/docs/development.md#updating-api-documentation))
- Has persistent data change

Side effects

- Possible performance regression
- Increased code complexity
- Breaking backward compatibility

Related changes

- PR to update [`pingcap/docs`](https://github.com/pingcap/docs)/[`pingcap/docs-cn`](https://github.com/pingcap/docs-cn):
- PR to update [`pingcap/tiup`](https://github.com/pingcap/tiup):
- Need to cherry-pick to the release branch

### Release note

```release-note
None.
```

### CODE REVIEW

File: tools/pd-api-bench/main.go [hunk 1]
```diff
@@ -39,18 +46,40 @@ import (
...
+	prometheus.MustRegister(pdAPIExecutionHistogram)
+	prometheus.MustRegister(pdAPIRequestCounter)
+
+	ctx, cancel := context.WithCancel(context.Background())
 	flagSet := flag.NewFlagSet("api-bench", flag.ContinueOnError)
 	flagSet.ParseErrorsWhitelist.UnknownFlags = true
+	flagSet.Int64Var(&qps, "qps", 1, "qps")
+	flagSet.Int64Var(&burst, "burst", 1, "burst")
+	flagSet.StringVar(&httpCases, "http-cases", "", "http api cases")
+	flagSet.StringVar(&gRPCCases, "grpc-cases", "", "grpc cases")
```
Comment (rleungx) [inline tools/pd-api-bench/main.go:82]: Do we need them?

File: tools/pd-api-bench/cases/controller.go [hunk 2]
```diff
@@ -0,0 +1,268 @@
...
+		gRPCClients: gRPCClients,
+		http:        make(map[string]*httpController),
+		grpc:        make(map[string]*gRPCController),
+	}
+}
+
+// GetHTTPCase returns the HTTP case config.
+func (c *Coordinator) GetHTTPCase(name string) (*Config, error) {
+	c.mu.Lock()
+	defer c.mu.Unlock()
```
Comment (rleungx) [inline tools/pd-api-bench/cases/controller.go:58]: Rlock?

File: tools/pd-api-bench/cases/controller.go [hunk 3]
```diff
@@ -0,0 +1,268 @@
...
+		pctx:     ctx,
+		clients:  clis,
+		HTTPCase: fn(),
+	}
+	return c
+}
+
+// run tries to run the HTTP api bench.
+func (c *httpController) run() {
+	if c.GetQPS() <= 0 || c.cancel != nil {
```
Comment (rleungx) [inline tools/pd-api-bench/cases/controller.go:162]: How about also validating it when setting it?

File: tools/pd-api-bench/main.go (hunk 1 above)
Comment (CabinfeverB) [reply tools/pd-api-bench/main.go:82]: This is to keep the original command line arguments, which is more convenient for simple testing

File: tools/pd-api-bench/cases/controller.go (hunk 3 above)
Comment (CabinfeverB) [reply tools/pd-api-bench/cases/controller.go:162]: This is a lazy implementation. In this way, the operation of adding, modifying, and deleting an API is implemented.

File: tools/pd-api-bench/cases/controller.go [hunk 4]
```diff
@@ -0,0 +1,268 @@
...
+		return
+	}
+	c.ctx, c.cancel = context.WithCancel(c.pctx)
+	qps := c.GetQPS()
+	burst := c.GetBurst()
+	cliNum := int64(len(c.clients))
+	tt := time.Duration(base/qps*burst*cliNum) * time.Microsecond
+	log.Info("begin to run gRPC case", zap.String("case", c.Name()), zap.Int64("qps", qps), zap.Int64("burst", burst), zap.Duration("interval", tt))
+	for _, cli := range c.clients {
```
Comment (rleungx) [inline tools/pd-api-bench/cases/controller.go:236]: In this way, we need to create the goroutine every time.

**Pull request merged**